### ISSUE TITLE
TypeInfo

### ISSUE THREAD

**Issue by burdges** [w3f/ark-scale#4]
We should define [`TypeInfo`](https://docs.rs/scale-info/latest/scale_info/) for arkworks types too, but in general cryptographic should be considered opaque fixed length byte arrays.  Assuming we've defined `MaxEncodeLen` their [`Type<MetaForm>`](https://docs.rs/scale-info/latest/scale_info/struct.Type.html) looks roughly like
```
impl scale_info::TypeInfo for $t {
    type Identity = Self;
    fn type_info() -> scale_info::Type {
        let type_def = scale_info::TypeDef {
            len: <Self as MaxEncodeLen>::max_encode_len(),
            type_param: scale_info::form::MetaForm,
        };
        scale_info::Type<MetaForm> {
            path: vec[String::from( core::any::type_name::<Self>() )],
            type_params: Vec::new(),  // no exposed type parameters.  
            type_def: scale_info::TypeDef::Array(type_def),
            docs: Vec::new(),  // optional
        }
    }
}
```

**Comment by burdges** [comment]
@davxy  What do you put into TypeInfo in the bandersnatch_vrfs substrate code now?

**Comment by davxy** [comment]
Most of the structs are maintained serialized (e.g. the signature).
Thus I implement type info just using the `#[derive(TypeInfo)]` on the newtype wrapping the byte array.

**Comment by burdges** [comment]
Closed by #6
